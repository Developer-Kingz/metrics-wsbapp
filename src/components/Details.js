import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

const Details = ({ country }) => {
  const { name, flags } = country;
  const { common } = name;
  return (
    <NavLink to={`/details/${common}`} className="navlink">
      <div className="details">
        <div className="details-inner">
          <img src={flags.png} alt="" className="flags" />
          <h4>{name.official}</h4>
        </div>
      </div>
    </NavLink>
  );
};

export default Details;

Details.propTypes = {
  country: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.number),
    PropTypes.arrayOf(PropTypes.string),
  ]).isRequired,
};
