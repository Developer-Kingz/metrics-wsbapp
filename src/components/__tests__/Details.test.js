import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Details from '../Details';

const mockCountry = {
  name: {
    official: 'Country Name',
  },
  flags: {
    png: 'flag.png',
  },
};

describe('Details component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Details country={mockCountry} />
      </BrowserRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
