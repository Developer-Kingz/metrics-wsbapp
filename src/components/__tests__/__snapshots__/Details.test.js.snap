// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="navlink"
    href="/details/undefined"
  >
    <div
      class="details"
    >
      <div
        class="details-inner"
      >
        <img
          alt=""
          src="flag.png"
        />
        <h4>
          Country Name
        </h4>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
